### ISSUE TITLE
objeckt disappeared after swap

### ISSUE THREAD

**Issue by tavaresartist** [hicetnunc2000/hicetnunc#276]
I collected an Objkt an then I swap'd it and then it disappeared. 

OBJKT: https://www.hicetnunc.xyz/objkt/13296

**Comment by mescalina** [comment]
Same for me, I've read that the swap function for bought objkts is there just cause they are testing it to implement the secondary market. I hope they are not lost, looks like they'll be accessible again once the secondary market is implemented.

**Comment by codewithfeeling** [comment]
I just collected https://www.hicetnunc.xyz/objkt/15137 and am testing out secondary market features. It has disappeared from my collection.

My wallet ID: tz1YJvMiZyXnzvV9pxtAiuCFvaG7XoBZhbUQ

Here are the transactions:

![image](https://user-images.githubusercontent.com/262516/112772831-e4685100-902a-11eb-8e13-d243205263e6.png)

1. [Original purchase for 7tez](https://tzkt.io/ongd6yiC8dvccj9TNrgkCUH8zbMFHHE5PHvnKFpVrr4rFiw2CMj)
2. [Relisted for 20 tez](https://tzkt.io/onoRqjWtVoM8Gtg3a8WRsXon7Jk32eAS2nUxu6KtTVHHh56vnrj)

After transaction 2, the item is gone from my collection and my address no longer appears in the "collectors" list. The swap seems to be live on the OBJKT page, but the piece is in zombie land... any ideas on how I can restore it? I also can't cancel the swap because I can't see the piece.

**Comment by andrevenancio** [comment]
> I collected an Objkt an then I swap'd it and then it disappeared.
> 
> OBJKT: https://www.hicetnunc.xyz/objkt/13296

what is your wallet id?

**Comment by snowy1nl** [comment]
I have the same issue, swapped an objeckt, it's gone from my collection and it's nowhere to be found. To make it even more complicated, both Temple and Tezos explorer don't show the Objeckt ID, so no idea where it went. :-(

Wallet ID https://tzkt.io/tz1XmqF5gnTjgE6Cxu5CzaT5H7vXiFfihe95

**Comment by codewithfeeling** [comment]
I don’t think you need to worry. It’s just been sent to the secondary market so the smart contract “owns” the work. There’s just no way to view these in your profile right now but they are working on that.

So you have: creations / up for sale / collection

**Comment by crzypatchwork** [comment]
it's needed a third tab on the tz profile with information about swaps so it won't disappear

**Comment by presstube** [comment]
This feels like the same issue but as the creator of an OBJKT if you cancel a swap on an edition:

- that edition disappears from "collectors" list

- the entire OBJKT becomes hidden on "creations" page

Is this the same issue?

---

Here is an OBJKT where this is occurring for me:

https://www.hicetnunc.xyz/objkt/26596

**Comment by esnho** [comment]
Hello!

Same here, I'm sure that I own this objkt: https://www.hicetnunc.xyz/objkt/36655

But it doesn't show on my collection: https://www.hicetnunc.xyz/tz/tz1Xye21UorssHqTHEyMQso5EG95sssea9vN

I see that the objkt is now owned by: **KT1Hkg5qeNhfwpKW4fXvq7HGZB9z2EnmCCA9**
https://api.better-call.dev/v1/contract/mainnet/KT1RJ6PbjHpwc3M5rw5s2Nbmefwbuwbdxton/tokens/holders?token_id=36655

I noticed that swapped objkt appears in creator page but not in secondary collectors.

Thanks for your attention! :)

**Comment by esnho** [comment]
I further dug this, if I understand well this POST request

https://51rknuvw76.execute-api.us-east-1.amazonaws.com/dev/tz

Does not return swapped objects on the secondary market.

**Comment by DonKarlssonSan** [comment]
There is a third party tool that can display NFT's that you have put up for sale on the secondary market:
https://commodum.xyz/
I think that functionality should exist on hicetnunc

For me it would be the best if the NFT's I put up for secondary sale appeared both in my collection and in another new third tab. I mean I still own it even though I put it for sale. Why should I not be able to adore my art I still own in my collection!?

In the real world I still have the things in my house even though I put them up for sale on eBay/Craigslist. It's only when someone buys the thing I send it to them and THEN it disappears from my house.

**Comment by TJohanson82** [comment]
I purchased #121271 for 16 tezos on June 8 at 11:12:10AM and swapped it later for a secondary market at June 8 at 11:30:06. The item was listed all day June 9 and this morning on June 10th my item has not been sold and it is no longer listed for sale on the secondary market. In fact, the total pieces of art is 30 yet I count all all the owners and there is only 29...What happened to my artwork???  

https://www.hicetnunc.xyz/objkt/121271
https://tzkt.io/ooM41xPkVtYJcEH62BDzkhVMd9h2S2nkn5BTNaM6oePpDY7gfXz
https://tzkt.io/oo4o34J62jZaSwKi5XADBjt1zAavP2yh2NTSd9LKuVnFb47dzdX
